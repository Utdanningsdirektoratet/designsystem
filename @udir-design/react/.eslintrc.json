{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*", "storybook-static"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "@udir-design/react/tsconfig.*?.json",
          "@udir-design/react/tsconfig.storybook.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
