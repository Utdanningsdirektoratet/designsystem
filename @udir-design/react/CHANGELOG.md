## 0.1.0-1 (2024-10-22)

build: don't bundle depenencies


## 0.1.0-0 (2024-10-22)

### ğŸš€ Features

- add Button component ([8d698eb](https://github.com/Utdanningsdirektoratet/designsystem/commit/8d698eb))
