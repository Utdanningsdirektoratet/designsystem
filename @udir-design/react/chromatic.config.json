{
  "onlyChanged": true,
  "projectId": "Project:667e8f07bf467ff4403dfe77",
  "storybookBaseDir": "@udir-design/react",
  "zip": true
}
