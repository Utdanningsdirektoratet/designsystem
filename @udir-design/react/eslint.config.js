import { FlatCompat } from '@eslint/eslintrc';
// eslint-disable-next-line @nx/enforce-module-boundaries
import baseConfig from '../../eslint.config.js';
import storybook from 'eslint-plugin-storybook';
import eslintjs from '@eslint/js';
const { configs } = eslintjs;

const compat = new FlatCompat({
  baseDirectory: import.meta.dirname,
  recommendedConfig: configs.recommended,
});

export default [
  ...baseConfig,
  ...compat.extends('plugin:@nx/react'),
  ...storybook.configs['flat/recommended'],
  {
    ignores: ['!.storybook'],
  },
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
    languageOptions: {
      parserOptions: {
        project: [
          '@udir-design/react/tsconfig.*?.json',
          '@udir-design/react/tsconfig.storybook.json',
        ],
      },
    },
  },
  {
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    // Storybook & docs-specific overrides
    files: ['**/*.stories.{ts,tsx}', '**/{.storybook,demo,docs}/**/*.{ts,tsx}'],
    rules: {
      '@nx/enforce-module-boundaries': [
        'error',
        {
          allowCircularSelfDependency: true,
        },
      ],
    },
  },
];
