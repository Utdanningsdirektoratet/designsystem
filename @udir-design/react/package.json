{
  "name": "@udir-design/react",
  "repository": {
    "type": "git",
    "url": "github:Utdanningsdirektoratet/designsystem",
    "directory": "@udir-design/react"
  },
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT",
  "version": "0.0.0-semantically-released",
  "type": "module",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    },
    "./alpha": {
      "import": "./dist/alpha.js",
      "require": "./dist/alpha.cjs"
    },
    "./beta": {
      "import": "./dist/beta.js",
      "require": "./dist/beta.cjs"
    },
    "./html": {
      "types": "./dist/html.d.ts"
    },
    "./style.css": "./dist/style.css"
  },
  "files": [
    "./dist"
  ],
  "scripts": {
    "build": "vite build",
    "inject": "pnpm i",
    "lint:cycles": "pnpm madge --circular ./src/alpha.ts",
    "test": "vitest --project unit",
    "typecheck": "tsc --build",
    "dev": "storybook dev --port 6005",
    "build:storybook": "storybook build --stats-json",
    "serve:storybook": "pnpm dlx http-server ./storybook-static",
    "test:storybook": "pnpm exec playwright install --with-deps chromium && pnpm vitest --project storybook"
  },
  "dependencies": {
    "@digdir/designsystemet-react": "1.5.0",
    "@navikt/aksel-icons": "^7.30.1",
    "@udir-design/css": "workspace:*",
    "tslib": "^2.8.1"
  },
  "devDependencies": {
    "@storybook/addon-a11y": "^9.1.5",
    "@storybook/addon-docs": "^9.1.5",
    "@storybook/addon-vitest": "^9.1.5",
    "@storybook/react-vite": "^9.1.5",
    "@types/diffable-html": "^5.0.2",
    "@types/mdast": "^4.0.4",
    "@types/ramda": "^0.30.2",
    "@types/react": "^19.1.13",
    "@types/react-dom": "^19.1.9",
    "@types/remark-heading-id": "^1.0.0",
    "@vitejs/plugin-react-swc": "^4.0.1",
    "@vitest/browser": "^3.2.4",
    "@vitest/coverage-v8": "^3.2.4",
    "@vitest/ui": "^3.2.4",
    "axe-core": "^4.10.3",
    "axe-playwright": "^2.2.2",
    "chromatic": "^13.1.4",
    "clsx": "2.1.1",
    "diffable-html": "^6.0.1",
    "eslint-plugin-storybook": "^9.1.5",
    "hast-util-from-html": "^2.0.3",
    "hast-util-is-element": "^3.0.0",
    "hast-util-to-html": "^9.0.5",
    "highcharts": "^12.4.0",
    "highcharts-react-official": "^3.2.2",
    "markdown-to-jsx": "^7.7.13",
    "mdast-util-to-hast": "^13.2.0",
    "playwright": "^1.55.0",
    "ramda": "^0.31.3",
    "react": "^19.1.1",
    "react-dom": "^19.1.1",
    "react-hook-form": "^7.62.0",
    "react-style-stringify": "^1.2.0",
    "remark-gfm": "^4.0.1",
    "remark-heading-id": "^1.0.1",
    "remark-parse": "^11.0.0",
    "remark-stringify": "^11.0.0",
    "sass": "^1.92.1",
    "storybook": "^9.1.5",
    "storybook-addon-pseudo-states": "^9.1.5",
    "storybook-addon-tag-badges": "^2.0.2",
    "unified": "^11.0.5",
    "unist-util-remove": "^4.0.0",
    "unist-util-visit": "^5.0.0",
    "vite": "^7.1.5",
    "vite-plugin-dts": "^4.5.4",
    "vite-tsconfig-paths": "^5.1.4",
    "vitest": "^3.2.4"
  },
  "peerDependencies": {
    "@types/react": "^18.3.7 || ^19.0.0",
    "react": "^18.3.1 || ^19.0.0",
    "react-dom": "^18.3.1 || ^19.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  }
}
