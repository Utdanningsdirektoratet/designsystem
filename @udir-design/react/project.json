{
  "name": "@udir-design/react",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "@udir-design/react/src",
  "projectType": "library",
  "tags": [],
  "// targets": "to see all targets run: nx show project @udir-design/react --web",
  "targets": {
    "dev": {
      "continuous": true,
      "dependsOn": [
        "build",
        { "projects": "!css", "target": "^watch" },
        "lint:cycles"
      ]
    },
    "build:storybook": {
      "inputs": ["default", "^default", "{workspaceRoot}/README.md"],
      "dependsOn": ["build"],
      "cache": true,
      "outputs": ["{projectRoot}/storybook-static"]
    },
    "serve:storybook": {
      "continuous": true,
      "dependsOn": ["build:storybook"]
    },
    "test:storybook": {
      "cache": true,
      "dependsOn": ["build"]
    }
  }
}
