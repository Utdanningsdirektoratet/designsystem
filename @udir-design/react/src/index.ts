export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
