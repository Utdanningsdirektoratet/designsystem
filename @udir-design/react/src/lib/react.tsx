import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
