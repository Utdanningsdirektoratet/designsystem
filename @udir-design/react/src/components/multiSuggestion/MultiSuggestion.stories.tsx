import type { Meta, StoryObj } from '@storybook/react-vite';
import { userEvent, within } from 'storybook/test';
import { type ChangeEvent, useState } from 'react';
import {
  MultiSuggestion,
  Button,
  Divider,
  Field,
  Label,
  Paragraph,
  Spinner,
} from '@udir-design/react/alpha';
import { useDebounceCallback } from '@digdir/designsystemet-react';

type Story = StoryObj<typeof MultiSuggestion>;

export default {
  component: MultiSuggestion,
  tags: ['alpha'],
  parameters: {
    layout: 'centered',
    customStyles: {
      /* add height by default */
      minHeight: '500px',
    },
    a11y: {
      // TODO: this rule should be enabled after https://github.com/dequelabs/axe-core/issues/4672 have propagated to @storybook/addon-a11y.
      config: {
        rules: [
          {
            id: 'aria-allowed-role',
            enabled: false,
          },
        ],
      },
    },
  },
  play: async (ctx) => {
    const storyRoot = ctx.canvasElement;
    // Refactored out the play function for easier reuse in the InModal story
    await testMultiSuggestion(storyRoot);
  },
} satisfies Meta;

async function testMultiSuggestion(el: HTMLElement) {
  /* When in test mode, open MultiSuggestion by focusing input */
  const input = within(el).getByRole('combobox');
  await userEvent.click(input);
}

const DATA_PLACES = [
  'Sogndal',
  'Oslo',
  'Stavanger',
  'Brønnøysund',
  'Trondheim',
  'Bergen',
];

export const Preview: Story = {
  render: (args, ctx) => {
    return (
      <MultiSuggestion {...args}>
        <MultiSuggestion.Chips />
        <MultiSuggestion.Input id={ctx.id} />
        <MultiSuggestion.Clear />
        <MultiSuggestion.List>
          <MultiSuggestion.Empty>Tomt</MultiSuggestion.Empty>
          {DATA_PLACES.map((place) => (
            <MultiSuggestion.Option key={place} value={place}>
              {place}
              <div>Kommune</div>
            </MultiSuggestion.Option>
          ))}
        </MultiSuggestion.List>
      </MultiSuggestion>
    );
  },
};

export const Controlled: Story = {
  render: (args, ctx) => {
    const [value, setValue] = useState<string[]>(['Oslo']);

    return (
      <>
        <Field>
          <Label>Velg reisemål du vil besøke</Label>
          <MultiSuggestion {...args} value={value} onValueChange={setValue}>
            <MultiSuggestion.Chips
              render={(e) => {
                return e.text;
              }}
            />
            <MultiSuggestion.Input id={ctx.id} />
            <MultiSuggestion.Clear />
            <MultiSuggestion.List>
              <MultiSuggestion.Empty>Tomt</MultiSuggestion.Empty>
              {DATA_PLACES.map((place) => (
                <MultiSuggestion.Option key={place} value={place}>
                  {place}
                  <div>Kommune</div>
                </MultiSuggestion.Option>
              ))}
            </MultiSuggestion.List>
          </MultiSuggestion>
        </Field>
        <Divider style={{ marginTop: 'var(--ds-size-4)' }} />

        <Paragraph style={{ margin: 'var(--ds-size-2) 0' }}>
          Valgte reisemål: {value.join(', ')}
        </Paragraph>

        <Button
          onClick={() => {
            setValue(['Sogndal', 'Stavanger']);
          }}
        >
          Sett reisemål til Sogndal, Stavanger
        </Button>
      </>
    );
  },
};

export const CustomFilterAlt1: Story = {
  render: (args, ctx) => {
    return (
      <Field>
        <Label>Skriv inn et tall mellom 1-6</Label>
        <MultiSuggestion
          {...args}
          filter={({ index, input }) => {
            console.log(!input.value || index === Number(input.value) - 1);
            return !input.value || index === Number(input.value) - 1;
          }}
        >
          <MultiSuggestion.Chips />
          <MultiSuggestion.Input id={ctx.id} />
          <MultiSuggestion.Clear />
          <MultiSuggestion.List>
            <MultiSuggestion.Empty>Tomt</MultiSuggestion.Empty>
            {DATA_PLACES.map((text) => (
              <MultiSuggestion.Option key={text} value={text.toLowerCase()}>
                {text}
              </MultiSuggestion.Option>
            ))}
          </MultiSuggestion.List>
        </MultiSuggestion>
      </Field>
    );
  },
};

export const CustomFilterAlt2: Story = {
  render: (args, ctx) => {
    const [value, setValue] = useState('');

    return (
      <Field>
        <Label>Skriv inn et tall mellom 1-6</Label>
        <MultiSuggestion {...args} filter={false}>
          <MultiSuggestion.Chips />
          <MultiSuggestion.Input
            id={ctx.id}
            value={value}
            onChange={(event) => setValue(event.target.value)}
          />
          <MultiSuggestion.Clear />
          <MultiSuggestion.List>
            <MultiSuggestion.Empty>Tomt</MultiSuggestion.Empty>
            {DATA_PLACES.filter(
              (_, index) => !value || index === Number(value) - 1,
            ).map((text) => (
              <MultiSuggestion.Option key={text}>{text}</MultiSuggestion.Option>
            ))}
          </MultiSuggestion.List>
        </MultiSuggestion>
      </Field>
    );
  },
};

export const AlwaysShowAll: Story = {
  render: (args, ctx) => {
    const [value, setValue] = useState('Sogndal');

    return (
      <Field>
        <Label>Viser alle options også når valgt</Label>
        <MultiSuggestion {...args} filter={false}>
          <MultiSuggestion.Chips />
          <MultiSuggestion.Input
            id={ctx.id}
            value={value}
            onChange={(event) => setValue(event.target.value)}
          />
          <MultiSuggestion.Clear />
          <MultiSuggestion.List>
            <MultiSuggestion.Empty>Tomt</MultiSuggestion.Empty>
            {DATA_PLACES.map((place) => (
              <MultiSuggestion.Option key={place}>
                {place}
              </MultiSuggestion.Option>
            ))}
          </MultiSuggestion.List>
        </MultiSuggestion>
      </Field>
    );
  },
};

export const FetchExternal: Story = {
  render: (args, ctx) => {
    const [value, setValue] = useState('');
    const [options, setOptions] = useState<string[] | null>(null);

    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
      const isTyping = (event.nativeEvent as InputEvent).inputType;
      const value = encodeURIComponent(event.target.value.trim());
      setValue(event.target.value);

      if (!isTyping) return; // Prevent API call if clicking on items in list
      setOptions(null); // Clear options

      if (!value) return;

      debounced(value);
    };

    const apiCall = async (value: string) => {
      const api = `https://restcountries.com/v2/name/${value}?fields=name`;
      const countries = await (await fetch(api)).json();
      setOptions(
        Array.isArray(countries) ? countries.map(({ name }) => name) : [],
      );
    };

    const debounced = useDebounceCallback(apiCall, 500);

    return (
      <Field lang="en">
        <Label>Search for countries (in english)</Label>
        <MultiSuggestion {...args} filter={false}>
          <MultiSuggestion.Chips />
          <MultiSuggestion.Input
            id={ctx.id}
            value={value}
            onChange={handleChange}
          />
          <MultiSuggestion.Clear />
          <MultiSuggestion.List singular="%d country" plural="%d countries">
            {!!value && (
              <MultiSuggestion.Empty>
                {options ? (
                  'Ingen treff'
                ) : (
                  <span
                    style={{ display: 'flex', alignItems: 'center', gap: 8 }}
                  >
                    <Spinner aria-hidden="true" data-size="sm" /> Laster...
                  </span>
                )}
              </MultiSuggestion.Empty>
            )}
            {options?.map((option) => (
              <MultiSuggestion.Option key={option}>
                {option}
              </MultiSuggestion.Option>
            ))}
          </MultiSuggestion.List>
        </MultiSuggestion>
      </Field>
    );
  },
};
