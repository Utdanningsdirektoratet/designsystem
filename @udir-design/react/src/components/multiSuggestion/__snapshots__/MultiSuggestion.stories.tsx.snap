// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Always Show All 1`] = `
<div>
  <label
    data-weight="medium"
    for="components-multisuggestion--always-show-all"
    id=":label4"
  >
    Viser alle options også når valgt
  </label>
  <u-tags
    role="group"
    aria-label="Viser alle options også når valgt"
  >
    <input
      type="text"
      list=":r4:"
      placeholder
      id="components-multisuggestion--always-show-all"
      value="Sogndal"
      aria-label="Viser alle options også når valgt, No selected"
      popovertarget=":r4:"
      aria-autocomplete="list"
      aria-controls=":r4:"
      aria-expanded="true"
      autocomplete="off"
      role="combobox"
    >
    <button
      data-icon="true"
      data-variant="tertiary"
      type="reset"
      aria-label="Tøm"
    >
    </button>
    <u-datalist
      data-sr-singular="%d forslag"
      data-sr-plural="%d forslag"
      id=":r4:"
      role="listbox"
      tabindex="-1"
      data-multiselectable="true"
      aria-labelledby=":label4"
    >
      <u-option
        data-empty="true"
        role="none"
        tabindex="-1"
        aria-disabled="false"
        aria-selected="false"
        value="Tomt                                                                                                    Sogndal"
      >
        Tomt
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Sogndal                                                                                                    Sogndal"
      >
        Sogndal
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Oslo                                                                                                    Sogndal"
      >
        Oslo
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Stavanger                                                                                                    Sogndal"
      >
        Stavanger
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Brønnøysund                                                                                                    Sogndal"
      >
        Brønnøysund
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Trondheim                                                                                                    Sogndal"
      >
        Trondheim
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Bergen                                                                                                    Sogndal"
      >
        Bergen
      </u-option>
    </u-datalist>
  </u-tags>
</div>
`;

exports[`Controlled 1`] = `
<div>
  <label
    data-weight="medium"
    for="components-multisuggestion--controlled"
    id=":label1"
  >
    Velg reisemål du vil besøke
  </label>
  <u-tags
    role="group"
    aria-label="Velg reisemål du vil besøke"
  >
    <data
      data-removable="true"
      value="Oslo"
      aria-label="OsloKommune, Press to remove, 1 of 1"
      role="button"
      tabindex="-1"
    >
      OsloKommune
    </data>
    <input
      type="text"
      list=":r1:"
      placeholder
      id="components-multisuggestion--controlled"
      value
      aria-label="Velg reisemål du vil besøke, Navigate left to find 1 selected"
      popovertarget=":r1:"
      aria-autocomplete="list"
      aria-controls=":r1:"
      aria-expanded="true"
      autocomplete="off"
      role="combobox"
    >
    <button
      data-icon="true"
      data-variant="tertiary"
      type="reset"
      aria-label="Tøm"
    >
    </button>
    <u-datalist
      data-sr-singular="%d forslag"
      data-sr-plural="%d forslag"
      id=":r1:"
      role="listbox"
      tabindex="-1"
      data-multiselectable="true"
      aria-labelledby=":label1"
    >
      <u-option
        data-empty="true"
        role="none"
        tabindex="-1"
        aria-disabled="false"
        aria-selected="false"
      >
        Tomt
      </u-option>
      <u-option
        value="Sogndal"
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Sogndal
        <div>
          Kommune
        </div>
      </u-option>
      <u-option
        value="Oslo"
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="true"
        selected
      >
        Oslo
        <div>
          Kommune
        </div>
      </u-option>
      <u-option
        value="Stavanger"
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Stavanger
        <div>
          Kommune
        </div>
      </u-option>
      <u-option
        value="Brønnøysund"
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Brønnøysund
        <div>
          Kommune
        </div>
      </u-option>
      <u-option
        value="Trondheim"
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Trondheim
        <div>
          Kommune
        </div>
      </u-option>
      <u-option
        value="Bergen"
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Bergen
        <div>
          Kommune
        </div>
      </u-option>
    </u-datalist>
  </u-tags>
</div>
<hr
  aria-hidden="true"
  style="margin-top: var(--ds-size-4);"
>
<p
  data-variant="default"
  style="margin: var(--ds-size-2) 0;"
>
  Valgte reisemål: Oslo
</p>
<button
  data-variant="primary"
  type="button"
>
  Sett reisemål til Sogndal, Stavanger
</button>
`;

exports[`Custom Filter Alt 1 1`] = `
<div>
  <label
    data-weight="medium"
    for="components-multisuggestion--custom-filter-alt-1"
    id=":label2"
  >
    Skriv inn et tall mellom 1-6
  </label>
  <u-tags
    role="group"
    aria-label="Skriv inn et tall mellom 1-6"
  >
    <input
      type="text"
      list=":r2:"
      placeholder
      id="components-multisuggestion--custom-filter-alt-1"
      value
      aria-label="Skriv inn et tall mellom 1-6, No selected"
      popovertarget=":r2:"
      aria-autocomplete="list"
      aria-controls=":r2:"
      aria-expanded="true"
      autocomplete="off"
      role="combobox"
    >
    <button
      data-icon="true"
      data-variant="tertiary"
      type="reset"
      aria-label="Tøm"
    >
    </button>
    <u-datalist
      data-sr-singular="%d forslag"
      data-sr-plural="%d forslag"
      id=":r2:"
      role="listbox"
      tabindex="-1"
      data-multiselectable="true"
      aria-labelledby=":label2"
    >
      <u-option
        data-empty="true"
        role="none"
        tabindex="-1"
        aria-disabled="false"
        aria-selected="false"
        value="Tomt                                                                                                    "
      >
        Tomt
      </u-option>
      <u-option
        value="sogndal                                                                                                    "
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Sogndal
      </u-option>
      <u-option
        value="oslo                                                                                                    "
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Oslo
      </u-option>
      <u-option
        value="stavanger                                                                                                    "
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Stavanger
      </u-option>
      <u-option
        value="brønnøysund                                                                                                    "
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Brønnøysund
      </u-option>
      <u-option
        value="trondheim                                                                                                    "
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Trondheim
      </u-option>
      <u-option
        value="bergen                                                                                                    "
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
      >
        Bergen
      </u-option>
    </u-datalist>
  </u-tags>
</div>
`;

exports[`Custom Filter Alt 2 1`] = `
<div>
  <label
    data-weight="medium"
    for="components-multisuggestion--custom-filter-alt-2"
    id=":label3"
  >
    Skriv inn et tall mellom 1-6
  </label>
  <u-tags
    role="group"
    aria-label="Skriv inn et tall mellom 1-6"
  >
    <input
      type="text"
      list=":r3:"
      placeholder
      id="components-multisuggestion--custom-filter-alt-2"
      value
      aria-label="Skriv inn et tall mellom 1-6, No selected"
      popovertarget=":r3:"
      aria-autocomplete="list"
      aria-controls=":r3:"
      aria-expanded="true"
      autocomplete="off"
      role="combobox"
    >
    <button
      data-icon="true"
      data-variant="tertiary"
      type="reset"
      aria-label="Tøm"
    >
    </button>
    <u-datalist
      data-sr-singular="%d forslag"
      data-sr-plural="%d forslag"
      id=":r3:"
      role="listbox"
      tabindex="-1"
      data-multiselectable="true"
      aria-labelledby=":label3"
    >
      <u-option
        data-empty="true"
        role="none"
        tabindex="-1"
        aria-disabled="false"
        aria-selected="false"
        value="Tomt                                                                                                    "
      >
        Tomt
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Sogndal                                                                                                    "
      >
        Sogndal
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Oslo                                                                                                    "
      >
        Oslo
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Stavanger                                                                                                    "
      >
        Stavanger
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Brønnøysund                                                                                                    "
      >
        Brønnøysund
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Trondheim                                                                                                    "
      >
        Trondheim
      </u-option>
      <u-option
        tabindex="-1"
        role="option"
        aria-disabled="false"
        aria-selected="false"
        value="Bergen                                                                                                    "
      >
        Bergen
      </u-option>
    </u-datalist>
  </u-tags>
</div>
`;

exports[`Fetch External 1`] = `
<div lang="en">
  <label
    data-weight="medium"
    for="components-multisuggestion--fetch-external"
    id=":label5"
  >
    Search for countries (in english)
  </label>
  <u-tags
    role="group"
    aria-label="Search for countries (in english)"
  >
    <input
      type="text"
      list=":r5:"
      placeholder
      id="components-multisuggestion--fetch-external"
      value
      aria-label="Search for countries (in english), No selected"
      popovertarget=":r5:"
      aria-autocomplete="list"
      aria-controls=":r5:"
      aria-expanded="true"
      autocomplete="off"
      role="combobox"
    >
    <button
      data-icon="true"
      data-variant="tertiary"
      type="reset"
      aria-label="Tøm"
    >
    </button>
    <u-datalist
      data-sr-singular="%d country"
      data-sr-plural="%d countries"
      id=":r5:"
      role="listbox"
      tabindex="-1"
      data-multiselectable="true"
      aria-labelledby=":label5"
    >
    </u-datalist>
  </u-tags>
</div>
`;

exports[`Preview 1`] = `
<u-tags
  role="group"
  aria-label
>
  <input
    type="text"
    list=":r0:"
    placeholder
    id="components-multisuggestion--preview"
    value
    aria-label=", No selected"
    popovertarget=":r0:"
    aria-autocomplete="list"
    aria-controls=":r0:"
    aria-expanded="true"
    autocomplete="off"
    role="combobox"
  >
  <button
    data-icon="true"
    data-variant="tertiary"
    type="reset"
    aria-label="Tøm"
  >
  </button>
  <u-datalist
    data-sr-singular="%d forslag"
    data-sr-plural="%d forslag"
    id=":r0:"
    role="listbox"
    tabindex="-1"
    data-multiselectable="true"
    aria-labelledby
  >
    <u-option
      data-empty="true"
      role="none"
      tabindex="-1"
      aria-disabled="false"
      aria-selected="false"
    >
      Tomt
    </u-option>
    <u-option
      value="Sogndal"
      tabindex="-1"
      role="option"
      aria-disabled="false"
      aria-selected="false"
    >
      Sogndal
      <div>
        Kommune
      </div>
    </u-option>
    <u-option
      value="Oslo"
      tabindex="-1"
      role="option"
      aria-disabled="false"
      aria-selected="false"
    >
      Oslo
      <div>
        Kommune
      </div>
    </u-option>
    <u-option
      value="Stavanger"
      tabindex="-1"
      role="option"
      aria-disabled="false"
      aria-selected="false"
    >
      Stavanger
      <div>
        Kommune
      </div>
    </u-option>
    <u-option
      value="Brønnøysund"
      tabindex="-1"
      role="option"
      aria-disabled="false"
      aria-selected="false"
    >
      Brønnøysund
      <div>
        Kommune
      </div>
    </u-option>
    <u-option
      value="Trondheim"
      tabindex="-1"
      role="option"
      aria-disabled="false"
      aria-selected="false"
    >
      Trondheim
      <div>
        Kommune
      </div>
    </u-option>
    <u-option
      value="Bergen"
      tabindex="-1"
      role="option"
      aria-disabled="false"
      aria-selected="false"
    >
      Bergen
      <div>
        Kommune
      </div>
    </u-option>
  </u-datalist>
</u-tags>
`;
